src/Temporalio/Client/WorkflowHandle.cs

```diff
@@ -302,6 +302,189 @@ public record WorkflowHandle(
+                options);
+        }
+
+        /// <summary>
+        /// Start a workflow update via a call to a WorkflowUpdate attributed method.
+        /// </summary>
+        /// <typeparam name="TWorkflow">Workflow class type.</typeparam>
+        /// <typeparam name="TUpdateResult">Update result type.</typeparam>
+        /// <param name="updateCall">Invocation of workflow update method.</param>
+        /// <param name="options">Extra options.</param>
+        /// <returns>Workflow update handle.</returns>
+        /// <remarks>WARNING: Workflow update is experimental and APIs may change.</remarks>
+        public Task<WorkflowUpdateHandle<TUpdateResult>> StartUpdateAsync<TWorkflow, TUpdateResult>(
+            Expression<Func<TWorkflow, Task<TUpdateResult>>> updateCall,
+            WorkflowUpdateOptions? options = null)
+        {
+            var (method, args) = ExpressionUtil.ExtractCall(updateCall);
+            return StartUpdateAsync<TUpdateResult>(
+                Workflows.WorkflowUpdateDefinition.NameFromMethodForCall(method),
+                args,
+                options);
+        }
+
+        /// <summary>
+        /// Start a workflow update using its name.
+        /// </summary>
+        /// <param name="update">Name of the update.</param>
+        /// <param name="args">Arguments for the update.</param>
+        /// <param name="options">Extra options.</param>
+        /// <returns>Workflow update handle.</returns>
+        /// <remarks>WARNING: Workflow update is experimental and APIs may change.</remarks>
+        public Task<WorkflowUpdateHandle> StartUpdateAsync(
+            string update, IReadOnlyCollection<object?> args, WorkflowUpdateOptions? options = null) =>
+            StartUpdateAsync<ValueTuple>(update, args, options).ContinueWith<WorkflowUpdateHandle>(
+                t => t.Result, TaskScheduler.Current);
+
+        /// <summary>
+        /// Start a workflow update using its name.
+        /// </summary>
+        /// <typeparam name="TUpdateResult">Update result type.</typeparam>
+        /// <param name="update">Name of the update.</param>
+        /// <param name="args">Arguments for the update.</param>
+        /// <param name="options">Extra options.</param>
+        /// <returns>Workflow update handle.</returns>
+        /// <remarks>WARNING: Workflow update is experimental and APIs may change.</remarks>
+        public Task<WorkflowUpdateHandle<TUpdateResult>> StartUpdateAsync<TUpdateResult>(
+            string update, IReadOnlyCollection<object?> args, WorkflowUpdateOptions? options = null) =>
+            Client.OutboundInterceptor.StartWorkflowUpdateAsync<TUpdateResult>(new(
+                Id: Id,
+                RunId: RunId,
+                FirstExecutionRunId: FirstExecutionRunId,
+                Update: update,
+                Args: args,
+                Options: options,
+                Headers: null));
+
+        /// <summary>
+        /// Shortcut for
```

Comment:
Worth an extra sentence just to make it obvious
```suggestion
        /// Start a workflow update and wait for it to complete
        /// Shortcut for
```

Reply:
:+1: I was mimicking my terminology for `ExecuteWorkflow` (e.g. [here](https://dotnet.temporal.io/api/Temporalio.Client.ITemporalClientExtensions.html#Temporalio_Client_ITemporalClientExtensions_ExecuteWorkflowAsync_Temporalio_Client_ITemporalClient_System_String_System_Collections_Generic_IReadOnlyCollection_System_Object__Temporalio_Client_WorkflowOptions_) but I can add a sentence.

Reply:
Extra sentence added